src/cmap/wire_protocol/compression.ts

```diff
@@ -39,12 +39,18 @@ export function compress(
 ): void {
   const zlibOptions = {} as zlib.ZlibOptions;
   switch (self.options.agreedCompressor) {
-    case 'snappy':
+    case 'snappy': {
       if ('kModuleError' in Snappy) {
         return callback(Snappy['kModuleError']);
       }
-      Snappy.compress(dataToBeCompressed, callback);
+      const snappyResult = Snappy.compress(dataToBeCompressed, callback);
```

Comment:
Should we check function length here as well?

Reply:
not sure what you are asking

Reply:
We could assert the function length is what we expect it to be before assuming the last arg takes a callback, but might be brittle?

```js
(function(){}).length === 0
(function(a){}).length === 1
(function(a, b){}).length === 2
```

Reply:
so the danger I am seeing with assuming that the compress parameter takes a callback is that a future snappy might be updated to do something different with that second argument... it would be nice if we could determine the snappy version before calling the function

Reply:
I added some logic to check the dependency version, let me know what you think.